editor/gui/editor_quick_open_dialog.cpp

```diff
@@ -159,16 +162,142 @@ void EditorQuickOpenDialog::popup_dialog(const Vector<StringName> &p_base_types,
 	search_box->grab_focus();
 }
 
-void EditorQuickOpenDialog::ok_pressed() {
-	item_selected_callback.call(container->get_selected());
+void EditorQuickOpenDialog::popup_dialog_for_property(const Vector<StringName> &p_base_types, Object *p_obj, String p_path) {
+	ERR_FAIL_COND(p_base_types.is_empty());
+	directly_modify_property = true;
+	property_object = p_obj;
+	property_path = p_path;
 
-	container->save_selected_item();
+	// What the property value was before this Quick Load window was opened.
+	// We keep track of this so that if the user previews a different value,
+	// but then cancels out of the window, we can restore it.
+	current_property_value = property_object->get(property_path);
+
+	container->init(p_base_types);
+	get_ok_button()->set_disabled(container->has_nothing_selected());
+
+	set_title(get_dialog_title(p_base_types));
+	popup_centered_clamped(Size2(780, 650) * EDSCALE, 0.8f);
+	search_box->grab_focus();
+}
+
+void EditorQuickOpenDialog::ok_pressed() {
+	update_property();
 	container->cleanup();
 	search_box->clear();
 	hide();
 }
 
+void EditorQuickOpenDialog::selection_changed() {
+	if (!EDITOR_GET("filesystem/quick_open_dialog/instant_preview")) {
+		return;
+	}
+
+	// In order for Instant Preview to work, we need to directly set the property
+	// value of the target object to the newly selected resource.
+	preview_property();
+
+	container->save_selected_item();
+
+	// Required due to the way keyboard navigation between items is handled;
+	// it doesn't appear to use Godot's normal focus system. We may want to
+	// update it to use the standard focus/navigation system in the future.
```

Comment:
The search box should always be focused.
You can disable focus for the result items to avoid stealing it from the text field.